src/ol_infrastructure/applications/concourse/__main__.py

```diff
@@ -533,102 +606,151 @@
-    description="Launch template for deploying Concourse worker nodes",
-    iam_instance_profile=ec2.LaunchTemplateIamInstanceProfileArgs(
-        arn=concourse_instance_profile.arn,
-    ),
-    image_id=concourse_worker_ami.id,
-    block_device_mappings=[
-        ec2.LaunchTemplateBlockDeviceMappingArgs(
-            device_name="/dev/xvda",
-            ebs=ec2.LaunchTemplateBlockDeviceMappingEbsArgs(
-                volume_size=concourse_config.get_int("worker_disk_size") or 25,
-                volume_type=DiskTypes.ssd,
-                delete_on_termination=True,
-            ),
+
+############################################
+#     Worker Node IAM + EC2 Deployment     #
+############################################
+concourse_worker_instance_profiles = []
+
+for worker_def in concourse_config.get_object("workers") or []:
+    worker_class_name = worker_def["name"]
+
+    # Create IAM role + attach policies to it
+    concourse_worker_instance_role = iam.Role(
+        f"concourse-instance-role-worker-{worker_class_name}-{stack_info.env_suffix}",
+        assume_role_policy=json.dumps(
+            {
+                "Version": "2012-10-17",
+                "Statement": {
+                    "Effect": "Allow",
+                    "Action": "sts:AssumeRole",
+                    "Principal": {"Service": "ec2.amazonaws.com"},
+                },
+            }
+        ),
+        path="/ol-applications/concourse/role/",
+        tags=aws_config.tags,  # We will leave all the IAM resources with default tags.
+    )
+
+    for iam_policy_name in worker_def["iam_policies"] or []:
+        iam_policy_object = iam_policy_objects[iam_policy_name]
+        iam.RolePolicyAttachment(
+            f"concourse-instance-policy-worker-{worker_class_name}-policy-{iam_policy_name}-{stack_info.env_suffix}",
+            policy_arn=iam_policy_object.arn,
+            role=concourse_worker_instance_role.name,
         )
-    ],
-    vpc_security_group_ids=[
-        concourse_worker_security_group.id,
-        consul_security_groups["consul_agent"],
-    ],
-    instance_type=InstanceTypes[worker_instance_type].value,
-    key_name="oldevops",
-    tag_specifications=[
-        ec2.LaunchTemplateTagSpecificationArgs(
-            resource_type="instance",
-            tags=aws_config.merged_tags(
-                {"Name": f"concourse-worker-{stack_info.env_suffix}"}
-            ),
+
+    concourse_worker_instance_profile = iam.InstanceProfile(
+        f"concourse-instance-profile-worker-{worker_class_name}-{stack_info.env_suffix}",
+        role=concourse_worker_instance_role.name,
+        path="/ol-applications/concourse/profile/",
+    )
+
+    concourse_worker_instance_profiles.append(concourse_worker_instance_profile)
+
+    # Create EC2 resources
+    build_worker_user_data_partial = partial(
+        build_worker_user_data,
+        worker_def.get("concourse_team"),
+        worker_def.get("concourse_tags") or [],
```

Comment:
Because `worker_def` is a dictionary you can pass the `[]` as the default value if there is no key at `"concourse_tags"`. So, `worker_def.get("concourse_tags", [])`

Reply:
👍